preprocessing/speaking_status/annotations_v2/analyse_and_plot.py

```diff
@@ -0,0 +1,204 @@
+                bottom=np.zeros_like(x_idx) + ((num_participants - j - 1) * 0.5),
+            )
+
+        ax.set_ylim(0, len(all_annotation_data) * 0.5)
+
+        # Save at 1090x1080 resolution
+        # fig.savefig(save_dir / f'plot_{i}_{end_idx}.png', dpi=100)
+        if multithreaded:
+            thread = threading.Thread(
+                target=savefig, args=(save_dir, fig, ax, i, end_idx)
+            )
+            thread.start()
+            thread_list.append(thread)
+        else:
+            savefig(save_dir, fig, ax, i, end_idx)
+
+    for thread in tqdm(thread_list, desc="thread"):
+        thread.join()
+
+
+def majority_vote(all_annotation_data: np.ndarray, valid_ids: list[int]) -> np.ndarray:
+    majority_vote = []
+    for i in range(all_annotation_data.shape[1]):
+        if i in valid_ids:
+            majority_vote.append(all_annotation_data[i, :])
+    majority_vote = np.array(majority_vote)
+    majority_vote = np.sum(majority_vote, axis=0)
+    result = np.zeros_like(majority_vote)
+    result[majority_vote > len(valid_ids) / 2] = 1
+    return result
+
+
+@dataclass
+class AggrementData:
+    data: list[float]
+    mean: Optional[float] = None
+    std: Optional[float] = None
+    num_marked_missing: int = 0
+
+
+def get_all_data_for_participant(
+    participant_id: int,
+    all_annotation_data: list[AnnotationData],
+    total_agreement: dict[str, AggrementData],
+) -> tuple[np.ndarray, np.ndarray, dict[str, AggrementData]]:
+    total_agreement = deepcopy(total_agreement)
+    participant_annotations = []
+    prolific_ids = []
+    min_len = np.inf
+    for all_annotation_data_i in all_annotation_data:
+        for node in all_annotation_data_i.nodes.values():
+            for response in node.responses:
+                if len(response.annotations) > 0:
+                    participant_annotation = response.annotations[participant_id - 1]
+                    if len(participant_annotation) > 1:
+                        participant_annotations.append(participant_annotation)
+                        prolific_ids.append(response.prolific_id)
+                        min_len = min(min_len, len(participant_annotation))
+                    else:
+                        if response.prolific_id[0] not in total_agreement:
+                            total_agreement[response.prolific_id[0]] = AggrementData([])
+                        total_agreement[response.prolific_id[0]].num_marked_missing += 1
+                        print(
+                            f"participant {participant_id} marked missing by {response.prolific_id}"
+                        )
+
+    participant_annotations = [
+        participant_annotation[:min_len]
+        for participant_annotation in participant_annotations
+    ]
+    return np.array(prolific_ids), np.array(participant_annotations), total_agreement
+
+
+def main(
+    database_files: list[Path],
+    num_annotated_participants: int,
```

Comment:
Please document briefly the variables, or at least num_annotated_participants which is not obvious why it is needed. "This script is useful to conduct a pilot in prolific academic in which only a subset of CONFLAB participants were annotated and inter annotator agreement can be robustly computed, allowing to qualify good annotators from the pool of prolific. "